workspace/lib/lita/handlers/teamcity.rb

```diff
@@ -494,6 +511,31 @@ def fetch_builds(build_url)
         return data
       end
 
+      def fetch_successful_builds(build_id_wildcard, response)
+        response_str = ''
+        count_success = 0
+        maximum_builds = 1000
+        end_time_teamcity, start_time_teamcity = get_teamcity_date_range()
```

Comment:
There is way too much logic here.  Use the build Locator in the API.  Example:

https://build.ankicore.com/app/rest/builds/?locator=status:success,count:10,sinceDate:20171205T000000-0800